test/e2e/autoscale_test.go

```diff
@@ -205,12 +143,22 @@ func generateTrafficAtFixedRPS(ctx *testContext, rps int, duration time.Duration
 			successfulRequests++
 		}
 	}
+}
 
-	if successfulRequests != totalRequests {
-		return fmt.Errorf("error making requests for scale up. total = %d, errors = %d, expected 0",
-			totalRequests, totalRequests-successfulRequests)
-	}
-	return nil
+func generateTrafficAtFixedConcurrency(ctx *testContext, concurrency int, duration time.Duration, stopChan chan struct{}) error {
+	pacer := vegeta.ConstantPacer{} // Sends requests as quickly as possible, capped by MaxWorkers below.
+	attacker := vegeta.NewAttacker(vegeta.Timeout(duration), vegeta.Workers(uint64(concurrency)), vegeta.MaxWorkers(uint64(concurrency)))
```

Comment:
I tried this before with cmd line to hit a revision with fixed single pod (`autoscaler-go` app). I found that the concurrency the queue-proxy received (~20) was much lower than the `workers` number (30). Could you try it to see whether you have similar problem?

Reply:
How exactly have you tried? What was the rate you configured for instance?

Reply:
This was the command I used: `vegeta attack  -duration 2h -workers=30 --max-workers=30 -rate=10000000`

Reply:
The rate needs to be 0, which means "as fast as possible"

Reply:
Try this out:

```golang
package main

import (
	"fmt"
	"net/http"
	"net/http/httptest"
	"sync"
	"time"

	vegeta "github.com/tsenart/vegeta/lib"
)

func main() {
	targetConcurrency := 100

	currentConcurrency := 0
	maxConcurrency := 0
	var mux sync.Mutex
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		mux.Lock()
		currentConcurrency++
		if currentConcurrency > maxConcurrency {
			maxConcurrency = currentConcurrency
		}
		mux.Unlock()

		time.Sleep(10 * time.Millisecond)

		mux.Lock()
		currentConcurrency--
		mux.Unlock()
	}))
	defer server.Close()

	pacer := vegeta.ConstantPacer{} // Sends requests as quickly as possible, capped by MaxWorkers below.
	attacker := vegeta.NewAttacker(vegeta.Timeout(10*time.Second), vegeta.Workers(uint64(targetConcurrency)), vegeta.MaxWorkers(uint64(targetConcurrency)))
	targeter := vegeta.NewStaticTargeter(vegeta.Target{URL: server.URL})

	results := attacker.Attack(targeter, pacer, 1*time.Second, "test")
	for range results {
	}

	fmt.Println(maxConcurrency)
}
```